### PR TITLE
fix: catchall handler should be added after plugin application

### PR DESCRIPTION
**Pull request by jlipps**
somewhere along the way plugins were broken. Maybe a change in Express? Anyway, need to put the catchall handler at the very end, after any plugin might be registered.

### CODE REVIEW

File: lib/express/server.js [hunk 1]
```diff
@@ -50,6 +50,11 @@ async function server (opts = {}) {
       configureServer({app, addRoutes: routeConfiguringFunction, allowCors, basePath, plugins});
       await configureServerPlugins({app, httpServer, plugins});
 
+      // once all configurations and plugins have been applied, make sure to set up a catchall
+      // handler so that anything unknown 404s. But do this after everything else since we don't
+      // want to block plugins' ability to add routes if they want.
+      app.all('*', catch404Handler);
```
Comment (mykola-mokhnach) [inline lib/express/server.js:56]: I'm not quite sure why this change is needed. The actual plugin routes are added in line 105 (`configureServer` function)

Comment (jlipps) [reply lib/express/server.js:56]: it's because there are two ways for plugins to update routes. one is via the extra routes way, and one is 'manually' so to speak, by messing with the express app object. for example here: https://github.com/appium/appium-fake-plugin/blob/master/lib/plugin.js#L25-L28

Comment (mykola-mokhnach) [reply lib/express/server.js:56]: ok, makes sense. IMHO, I would only keep routes update as the only possible way to do that, because the other way is too implicit.

**Review by mykola-mokhnach — APPROVED** [review]

File: lib/express/server.js (hunk 1 above)
Comment (jlipps) [reply lib/express/server.js:56]: i understand that perspective. unfortunately since we are giving them the app object, they can do whatever they want with it, and it seems better to support that than have some cases fail silently in weird ways.

**Comment by jlipps** [comment]
published in 7.3.1
